python/cudf/cudf/core/column/column.py

```diff
@@ -399,14 +400,23 @@ def set_mask(self, value) -> Self:
                 dbuf.copy_from_host(value)
                 mask = as_buffer(dbuf)
 
-        return build_column(  # type: ignore[return-value]
-            data=self.data,
-            dtype=self.dtype,
-            mask=mask,
-            size=self.size,
-            offset=0,
-            children=self.children,
+        if mask is not None:
+            new_mask: plc.gpumemoryview | None = plc.gpumemoryview(mask)
+            new_null_count = plc.null_mask.null_count(
+                new_mask,
+                0,
+                self.size,
+            )
+        else:
+            new_mask = None
+            new_null_count = 0
+        exposed = isinstance(self.data, ExposureTrackedBuffer)
```

Comment:
I'm afraid I don't remember the exact semantics of the buffer classes right now to be sure, but is this condition not too broad? Can't you have an ExposureTrackedBuffer that is not currently exposed? I don't think the current code will result in incorrect results in any case, it might just be less efficient than ideal by forcing copies.